wagtail/contrib/modeladmin/helpers/search.py

```diff
@@ -28,39 +27,39 @@ def show_search_form(self):
 
 
 class DjangoORMSearchHandler(BaseSearchHandler):
-    def search_queryset(self, queryset, search_term, **extra_search_kwargs):
+    def search_queryset(self, queryset, search_term, distinct_applied, **extra_search_kwargs):
         if not search_term or not self.search_fields:
-            return queryset, False
+            return queryset
 
         if extra_search_kwargs:
             queryset = queryset.filter(**extra_search_kwargs)
 
-        use_distinct = False
         orm_lookups = ['%s__icontains' % str(search_field)
                        for search_field in self.search_fields]
         for bit in search_term.split():
             or_queries = [Q(**{orm_lookup: bit})
                           for orm_lookup in orm_lookups]
             queryset = queryset.filter(reduce(operator.or_, or_queries))
         opts = queryset.model._meta
-        for search_spec in orm_lookups:
-            if lookup_needs_distinct(opts, search_spec):
-                use_distinct = True
-                break
-        return queryset, use_distinct
+        if not distinct_applied:
+            for search_spec in orm_lookups:
+                if lookup_needs_distinct(opts, search_spec):
+                    return queryset.distinct()
+        return queryset
+
 
     @property
     def show_search_form(self):
         return bool(self.search_fields)
 
 
 class WagtailBackendSearchHandler(BaseSearchHandler):
-    def search_queryset(self, queryset, search_term,
+    def search_queryset(self, queryset, search_term, distinct_applied,
                         operator=None, order_by_relevance=True, partial_match=True, backend='default'):
```

Comment:
Instead of adding `distinct_applied` explicitly here, could we just add `**kwargs` to the end of this signature?